### PR TITLE
indexer: rm AddressManager, StateCommitmentChain

### PR DESCRIPTION
**Pull request by tuxcanfly**
**Description**
This PR removes types not required for indexing bedrock anymore - `AddressManager` `StateCommitmentChain`

**Metadata**
- Fixes ENG-2664

### CODE REVIEW

File: indexer/config.go [hunk 1]
```diff
@@ -121,18 +118,17 @@ type Config struct {
...
-		DBPort:                  ctx.GlobalUint64(flags.DBPortFlag.Name),
-		DBUser:                  ctx.GlobalString(flags.DBUserFlag.Name),
-		DBPassword:              ctx.GlobalString(flags.DBPasswordFlag.Name),
-		DBName:                  ctx.GlobalString(flags.DBNameFlag.Name),
+		BuildEnv:           ctx.GlobalString(flags.BuildEnvFlag.Name),
+		EthNetworkName:     ctx.GlobalString(flags.EthNetworkNameFlag.Name),
+		ChainID:            ctx.GlobalInt64(flags.ChainIDFlag.Name),
+		L1EthRpc:           ctx.GlobalString(flags.L1EthRPCFlag.Name),
+		L2EthRpc:           ctx.GlobalString(flags.L2EthRPCFlag.Name),
+		L2GenesisBlockHash: ctx.GlobalString(flags.L2GenesisBlockHashFlag.Name),
```
Comment (tynes) [inline indexer/config.go:126]: What do you think about removing the L2 genesis block hash and instead use an L1 chain id and L2 chain id flag for validation that its talking to the correct chains?

**Comment by tynes** [comment]
This just means that no new release of the indexer can come out until bedrock ships, which i think is fine

**Comment by tynes** [comment]
Looks like there is a linting error

**Review by mslipper — APPROVED** [review]
LGTM assuming you fix the linting issue.

File: indexer/config.go (hunk 1 above)
Comment (mslipper) [reply indexer/config.go:126]: +1 on this. Can be handled separately but it's easier to pass in a chain ID than a genesis hash.

Comment (tuxcanfly) [reply indexer/config.go:126]: OK, think it was being used to partially index the chain (flag should be called `L2StartBlockHashFlag`), but I don't see it being much useful outside testing. Will remove.

Comment (tuxcanfly) [reply indexer/config.go:126]: Will handle this in a new PR.

**Pull request merged**